packages/block-editor/src/hooks/layout.js

```diff
@@ -372,9 +376,7 @@ export const withLayoutStyles = createHigherOrderComponent(
 			layout?.inherit || layout?.contentSize || layout?.wideSize
 				? { ...layout, type: 'constrained' }
 				: layout || defaultBlockLayout || {};
-		const layoutClasses = hasLayoutBlockSupport
-			? useLayoutClasses( attributes, name )
-			: null;
+		const layoutClasses = useLayoutClasses( attributes, name );
```

Comment:
If this doesn't run in trunk (fallback condition), the additional useSetting and useSelect might be having an impact on performance?

Reply:
Yes. It's still strange how a few extra select calls have such an impact.

Reply:
I don't understand why this would happen because even in this function, we have a useSelect call a few lines earlier and a useSettings call a few lines below. Also all the colour/typograpthy filters have tons of useSelect and useSettings calls.

Reply:
Somethings you can try since the useSetting('layout') is already done here is to transform this hook to a function or something to avoid doing that useSetting twice and see the impact.

Reply:
Tried this in [2919bbb](https://github.com/WordPress/gutenberg/pull/48900/commits/2919bbbc83321dbf3a572508370426f00661aea1).

Makes sense, every time we call this hook we already have that info anyway.

Reply:
Also, this hook signature just changed in #45299, so good to change it now.

Reply:
So it's not the `useSetting` call... Mysterious!

Reply:
I'm tempted to temporarily remove the position hook changes so we can cancel that out.

Reply:
Ok, with the position hooks reverted, the performance result looks better.